### PR TITLE
Scheduling Appointments - API End Points

### PR DESCRIPTION
**Pull request by akroon3r**
In order for the front end to use the model/schema for appointments, a GET detail, GET list, POST, PUT and DELETE http method needed to be developed. These end-points follow standards for office ID checking throughout the rest of the bookings feature. These end-points were tested using postman to ensure that proper responses were made to the actions taken against the API.

### CODE REVIEW

File: api/app/resources/bookings/appointment/appointment_delete.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    @oidc.accept_token(require_token=True)
+    def delete(self, id):
+
+        csr = CSR.find_by_username(g.oidc_token_info["username"])
+
+        appointment = Appointment.query.filter_by(appointment_id=id)\
+                                       .filter_by(office_id=csr.office_id)\
+                                       .first_or_404()
+
+        if appointment.office_id != csr.office_id:
```
Comment (sjrumsby) [inline api/app/resources/bookings/appointment/appointment_delete.py]: This check is no longer needed.

File: api/app/resources/bookings/appointment/appointment_detail.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    @oidc.accept_token(require_token=True)
+    def get(self, id):
+
+        csr = CSR.find_by_username(g.oidc_token_info['username'])
+
+        try:
+            appointment = Appointment.query.filter_by(appointment_id=id).first_or_404()
+
+            if appointment.office_id != csr.office_id:
```
Comment (sjrumsby) [inline api/app/resources/bookings/appointment/appointment_detail.py]: This check is no longer needed.

File: api/app/resources/bookings/appointment/appointment_detail.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    appointment_schema = AppointmentSchema()
+
+    @oidc.accept_token(require_token=True)
+    def get(self, id):
+
+        csr = CSR.find_by_username(g.oidc_token_info['username'])
+
+        try:
+            appointment = Appointment.query.filter_by(appointment_id=id).first_or_404()
```
Comment (sjrumsby) [inline api/app/resources/bookings/appointment/appointment_detail.py]: This query needs to filter by office_id

File: api/app/resources/bookings/appointment/appointment_put.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    def put(self, id):
+
+        csr = CSR.find_by_username(g.oidc_token_info["username"])
+
+        json_data = request.get_json()
+
+        if not json_data:
+            return  {"message": "No input data received for updating an appointment"}
+
+        appointment = Appointment.query.filter_by(appointment_id=id).first_or_404()
```
Comment (sjrumsby) [inline api/app/resources/bookings/appointment/appointment_put.py]: This needs a filter by office_id

File: api/app/resources/bookings/appointment/appointment_put.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+        appointment, warning = self.appointment_schema.load(json_data, instance=appointment, partial=True)
+
+        if warning:
+            logging.warning("WARNING: %s", warning)
+            return {"message": warning}, 422
+
+        if appointment.office_id == csr.office_id:
```
Comment (sjrumsby) [inline api/app/resources/bookings/appointment/appointment_put.py]: This check will no longer be needed.

File: api/app/resources/bookings/appointment/appointment_delete.py (hunk 1 above)
Comment (akroon3r) [reply api/app/resources/bookings/appointment/appointment_delete.py]: done

File: api/app/resources/bookings/appointment/appointment_detail.py (hunk 2 above)
Comment (akroon3r) [reply api/app/resources/bookings/appointment/appointment_detail.py]: done

File: api/app/resources/bookings/appointment/appointment_detail.py (hunk 3 above)
Comment (akroon3r) [reply api/app/resources/bookings/appointment/appointment_detail.py]: done

File: api/app/resources/bookings/appointment/appointment_put.py (hunk 4 above)
Comment (akroon3r) [reply api/app/resources/bookings/appointment/appointment_put.py]: done

File: api/app/resources/bookings/appointment/appointment_put.py (hunk 5 above)
Comment (akroon3r) [reply api/app/resources/bookings/appointment/appointment_put.py]: done

**Comment by akroon3r** [comment]
Change request resolved, pushed up and ready for re-review.

**Pull request merged**